src/lm_buddy/jobs/model_clients.py

```diff
@@ -82,7 +84,52 @@ def predict(self, prompt):
         return self._tokenizer.batch_decode(generated_ids, skip_special_tokens=True)[0]
 
 
-class OpenAIModelClient(BaseModelClient):
+class APIModelClient(BaseModelClient):
+    """General model client for APIs."""
+
+    def __init__(self, config: VLLMCompletionsConfig):
+        self._config = config
+
+        hf_model_loader = HuggingFaceModelLoader()
+        self._engine = hf_model_loader.resolve_asset_path(config.inference.engine)
+        self._system = config.inference.system_prompt
+
+    @abstractmethod
+    def _chat_completion(
```

Comment:
since an abstract method doesn't usually contain implementation, we might want to take out the system prompt, or alternatively make it a non-abstract method

Reply:
+100 I removed the default value as it did not make a lot of sense anyway